sherpa/models/template.py

```diff
@@ -17,6 +18,65 @@
 #  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 #
 
+# pylint: disable=invalid-name
+
+"""Use data as model templates.
+
+Allow models to be created from data, where the model data,
+represented as a `sherpa.models.basic.TableModel`, can be associated
+with a set of parameters. The `TemplateModel` class only supports
+access to the parameter values, but the `InterpolatingTemplateModel`
+class adds interpolation of the parameter values (although this is
+class should not be used directly, and the `KNNInterpolator` class
+used instead).
+
+Examples
+========
+
+A set of three templates are set for the "scale" parameter with values
+of 5, 17, and 25. All templates have the same independent axis (of [5,
+15, 40]) and the scale=5 value is [1, 2, 3], the scale=17 is [3, 7,
+4], and scale=25 is [10, 15, 0].
```

Comment:
I'm trying to avoid any particularly-astro-specific example. I'm hoping that the reworked intro and example will make it clearer, but I am too close to it now to review it sensibly.

Reply:
This is confusing to me

Reply:
I check the formatted readthedoc text and it reads ok there - easier than in the code